core/lib/contracts/src/lib.rs

```diff
@@ -331,13 +340,35 @@ impl BaseSystemContracts {
         let hash = hash_bytecode(&bytecode);
 
         let default_aa = SystemContractCode {
-            code: bytes_to_be_words(bytecode),
+            code: bytes_to_be_words(bytecode.clone()),
             hash,
         };
 
+        // If evm simulator is not enabled, use the default account bytecode and hash.
+        let (evm_simulator_bytecode, evm_simulator_hash) =
```

Comment:
This adds the EVM simulator retroactively for all protocol versions. I'd prefer not to do that. See the overall comment for more discussion.